### ISSUE TITLE
DCT visualize

### ISSUE THREAD

**Issue by ztjsw** [megvii-research/SOLQ#4]
Hi there,
I'm trying to visualize the mask after idct, the code is:
```
save_dir = './vis'
gt_mask_len = 512
n_keep = 128
processor_dct = ProcessorDCT(n_keep=n_keep, gt_mask_len=gt_mask_len)
#mask = cv2.imread(img_path, 0).astype(np.float32)
mask = np.load('mask.npy')
new_mask = np.array((mask==1)).astype(np.float32)

new_mask = cv2.resize(new_mask, (gt_mask_len, gt_mask_len))
coeffs = cv2.dct(new_mask)
cv2.imwrite(os.path.join(save_dir, '{}_coeffs.png'.format(name.split('.')[0])), coeffs)

idct = np.zeros((gt_mask_len**2))
vectors = torch.from_numpy(coeffs).flatten()
vectors = vectors[torch.tensor(processor_dct.zigzag_table)]

idct[:n_keep] = vectors.cpu().numpy()
idct = processor_dct.inverse_zigzag(idct, gt_mask_len, gt_mask_len)
cv2.imwrite(os.path.join(save_dir, '{}_i_coeffs.png'.format(name.split('.')[0])), idct)
re_mask = cv2.idct(idct)
max_v = np.max(re_mask)
min_v = np.min(re_mask)
re_mask = np.where(re_mask>(max_v+min_v) / 2., 255, 0)
cv2.imwrite(os.path.join(save_dir, '{}_recover.png'.format(name.split('.')[0])), re_mask)


plt.figure(1)
plt.imshow(new_mask)
plt.figure(2)
plt.imshow(re_mask)
```

new_mask shows:
![image](https://user-images.githubusercontent.com/17254336/123060755-9a321880-d43d-11eb-8197-5eb24de2fe4f.png)

re_mask shows:
![image](https://user-images.githubusercontent.com/17254336/123060848-ad44e880-d43d-11eb-94fe-2b4a30e17556.png)

The re_mask looks different with the original mask.

I was wondering why this is and if there is anything I am doing wrong.

**Comment by dbofseuofhust** [comment]
> Hi there,
> I'm trying to visualize the mask after idct, the code is:
> 
> ```
> save_dir = './vis'
> gt_mask_len = 512
> n_keep = 128
> processor_dct = ProcessorDCT(n_keep=n_keep, gt_mask_len=gt_mask_len)
> #mask = cv2.imread(img_path, 0).astype(np.float32)
> mask = np.load('mask.npy')
> new_mask = np.array((mask==1)).astype(np.float32)
> 
> new_mask = cv2.resize(new_mask, (gt_mask_len, gt_mask_len))
> coeffs = cv2.dct(new_mask)
> cv2.imwrite(os.path.join(save_dir, '{}_coeffs.png'.format(name.split('.')[0])), coeffs)
> 
> idct = np.zeros((gt_mask_len**2))
> vectors = torch.from_numpy(coeffs).flatten()
> vectors = vectors[torch.tensor(processor_dct.zigzag_table)]
> 
> idct[:n_keep] = vectors.cpu().numpy()
> idct = processor_dct.inverse_zigzag(idct, gt_mask_len, gt_mask_len)
> cv2.imwrite(os.path.join(save_dir, '{}_i_coeffs.png'.format(name.split('.')[0])), idct)
> re_mask = cv2.idct(idct)
> max_v = np.max(re_mask)
> min_v = np.min(re_mask)
> re_mask = np.where(re_mask>(max_v+min_v) / 2., 255, 0)
> cv2.imwrite(os.path.join(save_dir, '{}_recover.png'.format(name.split('.')[0])), re_mask)
> 
> 
> plt.figure(1)
> plt.imshow(new_mask)
> plt.figure(2)
> plt.imshow(re_mask)
> ```
> 
> new_mask shows:
> ![image](https://user-images.githubusercontent.com/17254336/123060755-9a321880-d43d-11eb-8197-5eb24de2fe4f.png)
> 
> re_mask shows:
> ![image](https://user-images.githubusercontent.com/17254336/123060848-ad44e880-d43d-11eb-94fe-2b4a30e17556.png)
> 
> The re_mask looks different with the original mask.
> 
> I was wondering why this is and if there is anything I am doing wrong.

Hi, thanks for attention, in your settings, gt_mask_len = 512, n_keep = 128, n_keep is too small for 512x512 original mask, so there are more details lost. I suggested gt_mask_len = 128, n_keep = 256 as describled in paper.

**Comment by ztjsw** [comment]
Thanks!
